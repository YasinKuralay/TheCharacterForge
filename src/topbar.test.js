import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import TopBar from "./topbar.js";

it("renders the TopBar component without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<TopBar />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("matches the previous snapshot of the TopBar Component", () => {
    const TopBarComponentSnapshot = renderer.create(<TopBar />).toJSON();
    expect(TopBarComponentSnapshot).toMatchSnapshot();
});
