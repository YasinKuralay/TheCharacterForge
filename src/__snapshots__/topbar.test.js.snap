// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the previous snapshot of the TopBar Component 1`] = `
<div>
  <div
    id="topBarPlaceholder"
  />
  <div
    id="topBar"
  >
    <img
      id="topbarLogo"
      src="./logo.png"
      width="160px"
    />
    <div
      className="topBarComponent"
      onClick={[Function]}
    >
      <i
        className="fas fa-home fa-lg"
      />
      <span>
         Home
      </span>
    </div>
    <a
      href="/logout"
    >
      <div
        className="topBarComponent"
        id="logout"
      >
        <i
          className="fas fa-sign-out-alt fa-lg"
        />
        <span>
           Logout
        </span>
      </div>
    </a>
  </div>
</div>
`;
